packages/many_cpus_benchmarking/examples/shared_data_same_action.rs

```diff
@@ -0,0 +1,92 @@
+//! Example demonstrating multiple threads operating on shared data structure, performing the same action.
+//!
+//! This scenario shows two workers that both read from a shared `HashMap`. Both workers perform the
+//! same type of operation (reading), but they may be accessing the data from different memory
+//! regions depending on the work distribution mode selected.
+
+#![allow(missing_docs, reason = "No need for API documentation in example code")]
+
+use std::collections::HashMap;
+use std::hint::black_box;
+use std::sync::{Arc, RwLock};
+
+use criterion::{Criterion, criterion_group, criterion_main};
+use many_cpus_benchmarking::{Payload, WorkDistribution, execute_runs};
+
+criterion_group!(benches, entrypoint);
+criterion_main!(benches);
+
+fn entrypoint(c: &mut Criterion) {
+    // Both workers perform the same action (reading), so we don't need payload exchange modes
```

Comment:
The comment here is misleading. The "self" modes have each worker isolated, working independently - we cannot use those modes because we require collaboration (one initializes, the other does not - so the other has to wait for initialization, which is collaboration). Adjust the documentation and guides to make this more clear.